HLTrigger/JetMET/plugins/HLTScoutingPFProducerV2.cc

```diff
@@ -0,0 +1,293 @@
+void HLTScoutingPFProducerV2::produce(edm::StreamID sid, edm::Event &iEvent, edm::EventSetup const &setup) const {
+  using namespace edm;
+
+  //get vertices
+  Handle<reco::VertexCollection> vertexCollection;
+  std::unique_ptr<Run3ScoutingVertexCollection> outVertices(new Run3ScoutingVertexCollection());
+  if (iEvent.getByToken(vertexCollection_, vertexCollection)) {
+    for (auto const& vtx : *vertexCollection) {
+      outVertices->emplace_back(MiniFloatConverter::reduceMantissaToNbitsRounding(vtx.x(), mantissaPrecision_),
+                                MiniFloatConverter::reduceMantissaToNbitsRounding(vtx.y(), mantissaPrecision_),
+                                MiniFloatConverter::reduceMantissaToNbitsRounding(vtx.z(), mantissaPrecision_),
+                                MiniFloatConverter::reduceMantissaToNbitsRounding(vtx.zError(), mantissaPrecision_),
+                                MiniFloatConverter::reduceMantissaToNbitsRounding(vtx.xError(), mantissaPrecision_),
+                                MiniFloatConverter::reduceMantissaToNbitsRounding(vtx.yError(), mantissaPrecision_),
+                                vtx.tracksSize(),
+                                MiniFloatConverter::reduceMantissaToNbitsRounding(vtx.chi2(), mantissaPrecision_),
+                                vtx.ndof(),
+                                vtx.isValid());
+    }
+  }
+
+  //get rho
+  Handle<double> rho;
+  std::unique_ptr<double> outRho(new double(-999));
+  if (iEvent.getByToken(rho_, rho)) {
+    outRho = std::make_unique<double>(*rho);
+  }
+
+  //get MET
+  Handle<reco::PFMETCollection> metCollection;
+  std::unique_ptr<double> outMetPt(new double(-999));
+  std::unique_ptr<double> outMetPhi(new double(-999));
+  if (doMet_ && iEvent.getByToken(metCollection_, metCollection)) {
+    outMetPt = std::make_unique<double>(metCollection->front().pt());
+    outMetPhi = std::make_unique<double>(metCollection->front().phi());
+  }
+
+  //get PF candidates
+  Handle<reco::PFCandidateCollection> pfCandidateCollection;
+  std::unique_ptr<Run3ScoutingParticleV2Collection> outPFCandidates(new Run3ScoutingParticleV2Collection());
+  if (doCandidates_ && iEvent.getByToken(pfCandidateCollection_, pfCandidateCollection)) {
+    for (auto const& cand : *pfCandidateCollection) {
+      if (cand.pt() > pfCandidatePtCut_ && std::abs(cand.eta()) < pfCandidateEtaCut_) {
+        int vertex_index = -1;
+        int index_counter = 0;
+        double dr2 = 0.0001;
+        for (auto const& vtx : *outVertices) {
+          double tmp_dr2 = pow(vtx.x() - cand.vx(), 2) + pow(vtx.y() - cand.vy(), 2) + pow(vtx.z() - cand.vz(), 2);
+          if (tmp_dr2 < dr2) {
+            dr2 = tmp_dr2;
+            vertex_index = index_counter;
+          }
+          if (dr2 == 0.0)
+            break;
+          ++index_counter;
+        }
+        const auto *reco_cand = dynamic_cast<const reco::PFCandidate *>(&cand);
```

Comment:
Doesn't `cand` already have a type`const reco::PFCandidate&`? Why is this `dynamic_cast` needed?

Reply:
You're right, it is not necessary anymore. I'll remove it.

Reply:
Fixed it [here](https://github.com/cms-sw/cmssw/pull/36840/commits/f933d59922918b5c0c43b53e01f30c1c9233702c#diff-87d3bd14df755ce99831c3fbcd4d87245fd0cb4c11025301c20e4c4e8b6d2e2dL157).